### PR TITLE
Add more info about getting a REST API token

### PR DESCRIPTION
**Pull request by nylen**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -5,10 +5,21 @@ Compatible with Node.js and web browsers.
...
+- If you just want to create a token, the
+  [example app bundled with `wpcom-oauth`](https://github.com/Automattic/node-wpcom-oauth/tree/master/example)
+  is probably the quickest way.
+
 ### Node.js
 
-Introduce the `wpcom` dependency in your `package.json`, and
-then initialize it with your API Token.
+Introduce the `wpcom` dependency into your `package.json`, and then initialize
+it with your API token.
```
Comment (retrofox) [inline Readme.md]: Maybe we can add that `token` is optional and it's needed for actions that needs permissions (add/edit/delete a post, etc)

Comment (nylen) [inline Readme.md]: Done in 2eb3fcc.

Comment (retrofox) [inline Readme.md]: Sorry @nylen 
I meant just to say that `token` is an optional parameter:

```
... and then initialize it with your API token (optional).
```
The example must be very very simple.

Comment (nylen) [inline Readme.md]: If we're going to say that the API token is optional, shouldn't we provide some guidance about which situations require a token and which ones do not?

Comment (retrofox) [inline Readme.md]: Absolutely yes but in another section.

Comment (nylen) [inline Readme.md]: OK, how about moving these 2 examples (one that requires permissions & one that does not) to the **Examples** section?

Comment (retrofox) [inline Readme.md]: Yes, we have to think in useful sections. It could be examples page, or in the same API doc, etc

**Comment by nylen** [comment]
@retrofox the diff was getting ugly so I rebased.  Changes based on our discussion:

- Added new **Authentication** section with links about OAuth2 tokens.  These links are things I did not know when I started using `wpcom.js`, so I think they will be helpful to other users too.

- Added two simple examples in **Examples** section: one that requires a token and one that does not.

Does that help address your concerns about sections?  If not, please let me know what you'd like me to change.

File: Readme.md [hunk 2]
```diff
@@ -25,10 +26,26 @@ Include `dist/wpcom.js` in a `<script>` tag:
...
 </script>
 ```
 
+### Authentication
+
+Not all requests require a REST API token.  For example, listing posts on a
+public site is something anyone can do.
+
+If you do need a token, here are some links that will help you generate one:
+- [OAuth2 Authentication](https://developer.wordpress.com/docs/oauth2/)
```
Comment (retrofox) [inline Readme.md]: Add ` to this link like the other ones

Comment (nylen) [inline Readme.md]: The `code style` is for code: library names, functions, etc.  If you want these links to look consistent I'd rather remove the \` from every link there.

Comment (retrofox) [inline Readme.md]: It looks different. https://cloudup.com/cpVuGB4h3mK

**Pull request merged**